packages/terminal/src/browser/terminal-widget.ts

```diff
@@ -95,9 +95,10 @@ export class TerminalWidget extends BaseWidget {
 
     public async start(): Promise<void> {
         this.registerResize();
-        const root = await this.workspaceService.root;
+        const root = await this.workspaceService.tryRoot;
+        const rootUri = root !== undefined ? root.uri : undefined;
```

Comment:
I think it would be much better if you could pass the FS's `getCurrentUserHome` from the client. But I won't hold up this PR if others have no objections.

CC: @svenefftinge

Reply:
I thought the same, but I'm fine with both solutions.

Reply:
I thought about this but it would introduce a unnecessary dependency on filesystem.  Moreover i think this should come from the workspace service default.